### DISCUSSION TITLE
How to achieve the wireframe rendering feature

### DISCUSSION THREAD

**Discussion by Eagleton** [gfx-rs/wgpu#1818] [Q&A]
Well, at first I wanted to use geometry shader to achieve wireframe rendering, like https://github.com/Chaser324/unity-wireframe do. But I find that wgpu only has compute, vertex, and fragment shaders and I believe this is by design.  And then, I find PolygonMode in wgpu::PrimitiveState have Line, but it requires Features::NON_FILL_POLYGON_MODE to be enabled.
And the question is how to enable the Features::NON_FILL_POLYGON_MODE? 
If I can't do that, is there any way to do wireframe rendering?

**Comment by kvark** [comment]
You pass the features to `request_device`, which you want to enable. Passing any features is non-porable, and you need to check the adapter features first, technically. This is a feature in the first place because the native APIs do not guarantee it being available.

**Reply by Eagleton** [reply]
Well, I want to know if there is any example to learn

**Reply by kvark** [reply]
Take a look at the feature matrix of the examples  - https://github.com/gfx-rs/wgpu/tree/master/wgpu/examples#feature-matrix
The thing relevant here is "polygon mode", which is indicated for the cube example.
It returns the feature here - https://github.com/gfx-rs/wgpu/blob/c493b63b33ae84de03c2e9856b9b8c2f1e59d525/wgpu/examples/cube/main.rs#L111
The example framework then uses it here - https://github.com/gfx-rs/wgpu/blob/c493b63b33ae84de03c2e9856b9b8c2f1e59d525/wgpu/examples/framework.rs#L148

**Reply by Eagleton** [reply]
Thank you!

**Reply by Eagleton** [reply]
I also want to ask another question, is there a way to achieve transparent wireframe with shaders? It means the faces are transparent

**Reply by kvark** [reply]
Blending is something to look for. Again, see the feature matrix.
